doc/sslang-lrm.tex

```diff
@@ -715,7 +716,21 @@ \subsection{Sequences}
 
 
 \subsection{Pattern Matching}
-\fixme{todo}
+\begin{grammar}
+<match> ::=  "match" <expr_blk> "{" (  <pat> "=>" "{" <expr> "}" )$^{+}$ "}"
+\end{grammar}
+
+Pattern matches are a succinct way to handle of a variety of cases for the expression in the match clause, known as the scrutinee. If the scrutinee satisfies the pattern in a match arm, the expression in the match arm is evaluated; otherwise, the expression in that arm is not evaluated.
+
+The below code sample checks if the scrutinee \texttt{x} is \texttt{Some a}; if it is, \texttt{y} is set to \texttt{1}. Otherwise, if \texttt{x} is \texttt{None},\texttt{y} is set to \texttt{2}. Pattern matches are an elegant way to handle multiple cases of \texttt{if} statements. They also allow for the destruction of complex algebraic data types via comparison to the patterns in the match arms.
+
+\begin{lstlisting}
```

Comment:
I would take out the comments from this listing; they don't really add very much other than narrating the syntax. `y = 0` is also not valid grammar (you need a leading `let`), and even if it were, `y` cannot be reassigned.

How about just this:

```
match x
  Some t => t
  None => 0
 ```

This sample code scrutinizes `x`, an `Option Int`. It unwraps the `Option` type into a plain `Int`, and collapses `None` values into `0`.